bundles/org.openhab.binding.zoneminder/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,217 @@
+				<option value="Mocord">Mocord</option>
+				<option value="Nodect">Nodect</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="alarm">
+		<item-type>Switch</item-type>
+		<label>Alarm</label>
+		<description>Monitor alarm status</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
+	<channel-type id="state">
+		<item-type>String</item-type>
+		<label>State</label>
+		<description>Current monitor state</description>
+		<state readOnly="true" pattern="%s">
+			<options>
+				<option value="UNKNOWN">UNKNOWN</option>
+				<option value="IDLE">IDLE</option>
+				<option value="PREALARM">PREALARM</option>
+				<option value="ALARM">ALARM</option>
+				<option value="ALERT">ALERT</option>
+				<option value="TAPE">TAPE</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="triggerAlarm">
+		<item-type>Switch</item-type>
+		<label>Trigger Alarm</label>
+		<description>Triggers an alarm</description>
+		<state pattern="%s"></state>
+	</channel-type>
+	<channel-type id="events">
+		<item-type>Number</item-type>
+		<label>Number of Events</label>
+		<description>Number of events in time period</description>
+		<state readOnly="true" pattern="%.0f"></state>
+	</channel-type>
+	<channel-type id="url">
+		<item-type>String</item-type>
+		<label>URL</label>
+		<description>URL of image or stream</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
+	<channel-type id="eventId">
+		<item-type>String</item-type>
+		<label>Event Id</label>
+		<description>Id of the event</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
+	<channel-type id="eventName">
+		<item-type>String</item-type>
+		<label>Event Name</label>
+		<description>Name of the event</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
+	<channel-type id="eventCause">
+		<item-type>String</item-type>
+		<label>Event Cause</label>
+		<description>Cause of the event</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
+	<channel-type id="eventNotes">
+		<item-type>String</item-type>
+		<label>Event Notes</label>
+		<description>Notes for the event</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
+	<channel-type id="eventStart">
+		<item-type>DateTime</item-type>
+		<label>Event Start</label>
+		<description>Start date/time of the event</description>
+		<state readOnly="true" pattern="%1$tY-%1$tm-%1$td %1$tH:%1$tM:%1$tS"></state>
+	</channel-type>
+	<channel-type id="eventEnd">
+		<item-type>DateTime</item-type>
+		<label>Event End</label>
+		<description>End date/time of the event</description>
+		<state readOnly="true" pattern="%1$tY-%1$tm-%1$td %1$tH:%1$tM:%1$tS"></state>
+	</channel-type>
+	<channel-type id="eventFrames">
+		<item-type>Number</item-type>
+		<label>Event Frames</label>
+		<description>Number of frames in the event</description>
+		<state readOnly="true" pattern="%.0f"></state>
```

Comment:
I'm not sure that you can have fractional frames.
```suggestion
		<state readOnly="true" pattern="%d"></state>
```

Reply:
Correct. Fixed.